src/python/pants/option/global_options.py

```diff
@@ -406,6 +406,9 @@ def register_bootstrap_options(cls, register):
                   'Because pantsd currently does not support parallel runs, '
                   'any prior running Pants command must be finished for the current one to start. '
                   'To never timeout, use the value -1.')
+    register('--pantsd-local-client-fallback', type=bool,
```

Comment:
Consider renaming to `--pantsd-fallback-to-local-client` / `--no-pantsd-fallback-to-local-client`. 

The current wording made me expect this to be an enum option where you decide what behavior you want when falling back. Instead, the reword is meant to better express that this is a boolean option.